src/main/webapp/app/course/manage/course-management.component.ts

```diff
@@ -53,6 +56,37 @@ export class CourseManagementComponent implements OnInit, OnDestroy, AfterViewIn
             (res: HttpResponse<Course[]>) => {
                 this.courses = res.body!;
                 this.courseForGuidedTour = this.guidedTourService.enableTourForCourseOverview(this.courses, tutorAssessmentTour, true);
+                this.courseSemesters = this.courses
+                    .map((c) => c.semester ?? '')
+                    .filter((value, index, self) => {
+                        // filter down to unique values
+                        return !this.courses[index].testCourse && self.indexOf(value) === index;
+                    })
+                    .sort((a, b) => {
+                        if (a === '') {
+                            return 1;
+                        }
+                        if (b === '') {
+                            return -1;
+                        }
+                        const yearsCompared = parseInt(b.substr(2, 2), 10) - parseInt(a.substr(2, 2), 10);
+                        if (yearsCompared !== 0) {
+                            return yearsCompared;
+                        }
+                        // if years are the same, sort WS over SS
+                        return a.substr(0, 2) === 'WS' ? -1 : 1;
```

Comment:
Would this be a good candidate for a `compareBySemester` function in `Course`? This would also make this more compact.

Reply:
The problem is that we're already sorting over string values, not courses, so I cannot directly use a compare function on `Course`. Changing this to compare courses would be possible, but imo not ideal. The string values are already filtered, i.e. we're only dealing with a few distinct semester tags, which is better for performance than sorting over the entire courses list.